src/contexts/PreferencesContext.tsx

```diff
@@ -93,10 +93,13 @@ export const PreferencesProvider = ({
       ).then(response => response.json());
       return data?.value;
     } catch (error) {
-      console.log(
-        `Potential error fetching preferences, or preference with key ${key} is not set:`,
-        error
-      );
+      if (error instanceof HTTPError && error.responseCode === 404) {
+        console.log(`Preference '${key}' not found`);
+        return null;
+      }
+      else {
+        console.log(`Error fetching preference '${key}':`, error);
+      }
```

Comment:
what do you return in this branch? does it make sense to move the return out of the if?